src/processors/file-handler.js

```diff
@@ -112,8 +113,9 @@ export async function* readBatch(asyncIterator, fileName) {
       if (batch.jsonpath && batch.jsonpath.length > 1) {
```

Comment:
This is indeed a bit too messy. At minimum, loaders.gl should provide a helper function that takes the final batch and the accumulated `data` array and stich them together, handling all the corner cases. In preparation, I'd start by breaking out the logic here:

`rebuildJSONObject(finalBatch, accumulatedData)`